### PR TITLE
Moves MAD calculation out of plotting functions and into util.py

### LINKED ISSUE #227: Move MAD calculation to a separate lib/mad.py file
As we've discussed elsewhere (e.g. #155) we should move the MAD calculation to a separate file in the lib folder to reduce duplication

### PR DESCRIPTION
**Pull request by kevin218**
This resolves #227... I hope!

### CODE REVIEW

**Comment by kevin218** [comment]
Actually, let's hold off on reviewing until I make a few more changes

**Comment by kevin218** [comment]
Ok, ready for review

File: eureka/S3_data_reduction/s3_reduce.py [hunk 1]
```diff
@@ -343,10 +343,14 @@ def reduceJWST(eventlabel, ecf_path='./', s2_meta=None):
                 meta.tab_filename = meta.outputdir + 'S3_' + event_ap_bg + "_Table_Save.txt"
                 astropytable.savetable_S3(meta.tab_filename, time, wave_1d, stdspec, stdvar, optspec, opterr)
 
+            # Compute MAD alue
+            meta.mad_s3 = util.get_mad(meta, wave_1d, optspec)
```
Comment (taylorbell57) [inline eureka/S3_data_reduction/s3_reduce.py:347]: Bahaha, I love that function name

**Review by taylorbell57 — APPROVED** [review]

**Comment by taylorbell57** [comment]
Did you run pytest on this to confirm that everything still works?

**Pull request merged**